### PR TITLE
Fix EntityJoinWorldEvent not firing as expected:

### LINKED ISSUE #7962: [1.17.1] EntityJoinWorldEvent not firing on Server for existing entities
**Minecraft Version**: 1.17.1

**Forge Version**: 1.17.1-37.0.13

**Logs**: Not Applicable

**Steps to reproduce**:
Implement EntityJoinWorldEvent
Server side, log the names of every entity's class that joins
Load a save game / start a server
Only the player and newly spawned entities show up

**Description**:
The event only fires for newly created entities where the documentation suggests it should also fire when saved entities are loaded for the first time. It does fire in this way on the client and did in 1.16.5 on both sides. Wanted to check if this is an intended change or a bug.

### LINKED ISSUE #8030: [1.17.1] Entity Join World Event not firing on all mobs
Some mobs don't seem to fire the entity join world event.
The mobs I noticed were: Sheep, Cow, Chicken, Pig, Horse, Villager, Iron Golem, End Crystal
However, if I spawn them via spawn egg, they trigger the event.

I would've sent the log file, but it is veerryy messy from my mod.

### PR DESCRIPTION
**Pull request by gigaherz**
- Fix entities loaded from disk not getting the event.
- Fix vehicles with passengers getting the event twice.

Fixes #7962 
Fixes #8030

### CODE REVIEW

File: patches/minecraft/net/minecraft/world/level/entity/PersistentEntitySectionManager.java.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+       return this.m_157538_(p_157534_, false);
+    }
+ 
+-   private boolean m_157538_(T p_157539_, boolean p_157540_) {
++   public boolean addNewEntityWithoutEvent(T p_157534_) {
++      return this.addEntityWithoutEvent(p_157534_, false);
++   }
++
++   private boolean m_157538_(T p_157539_, boolean p_157540_)
++   {
```
Comment (pupnewfster) [inline patches/minecraft/net/minecraft/world/level/entity/PersistentEntitySectionManager.java.patch]: Bracket should be on previous line still

Comment (gigaherz) [reply patches/minecraft/net/minecraft/world/level/entity/PersistentEntitySectionManager.java.patch]: Oops.

**Comment by Xyndra** [comment]
What version did this get added to? I am using 37,0.45 and it isn't working

**Comment by gigaherz** [comment]
Uhm... this HASN'T been added... that's the point of a pull *request*. It's waiting for review.

**Comment by ricksouth** [comment]
What's preventing the merge? I've got a few mods depending on this which haven't worked in 1.17 so far. Looking forward to the fix 😇

**Review by marchermans — APPROVED** [review]

**Comment by ricksouth** [comment]
Thank you giga for the pull request, and Orion for approving the changes. Looking forward to the merge.

**Comment by jigglebi** [comment]
The installer gets an error every time I click on it.
Untitled

**Comment by TheCurle** [comment]
That's not even remotely related to this PR...
Go to the [forum](https://forums.minecraftforge.net) for further support.

**Pull request merged**